File: algorithms/python/Challenge_Two_City_Scheduling.py
class Solution:
    def twoCitySchedCost(self, costs: List[List[int]]) -> int:
        N = len(costs)
        
        costs.sort(key=lambda cost: cost[0] - cost[1])
        return sum([c[0] for c in costs[:N//2]]) + sum([c[1]  for c in costs[N//2:]])

File: algorithms/java/Challenge_Two_City_Scheduling.java
class Solution {
    public int twoCitySchedCost(int[][] costs) {
        Arrays.sort(costs, Comparator.comparingDouble(c -> c[0] - c[1]));
        
        int retVal= 0;
        int N = costs.length/2;
        for(int i = 0 ; i < N ; i++)
            retVal += costs[i][0] + costs[i+N][1];
        
        return retVal;
    }
}
